pages/en/publish/pwa-play-billing.md

```diff
@@ -0,0 +1,222 @@
+
+To prevent fraud, it’s critical to verify the purchase and purchase token on your back-end server. It’s also a good idea to keep track of users and their associated purchase tokens. Learn how to [implement the verification on your back-end server](/{{locale.code}}/publish/play-billing-backend).
+
+After validating the purchase, call `complete()` on the payment response to finish the payment flow and close out the billing UI. You can also pass in an optional `result` string to indicate the state of the payment process. It is up to the browser whether to provide any indication of this result to the user. Chrome does not create any user-visible cues so it is recommended that you display your own error or success messages in your PWA.
+
+```js
+const { purchaseToken } = paymentResponse.details;
+
+let paymentComplete;
+if (validatePurchaseOnBackend(purchaseToken)) {
+  paymentComplete = await paymentResponse.complete('success');
+  // Let user know their purchase transaction has successfully completed and been verified
+} else {
+  paymentComplete = await paymentResponse.complete('fail');
+  // Let user know their purchase transaction failed to verify
+}
+```
+
+### Subscription upgrades and downgrades
+
+This purchase flow is the same for both in-app products and subscription purchases. However, for subscriptions, Google Play has additional purchase options you can implement: upgrade and downgrade. When building the `data` for the payment method, you’ll need to pass in the following to initiate an upgrade or downgrade flow:
+
+- `sku`: This is the item ID for the new subscription to be upgraded or downgraded to.
+- `oldSku`: This is the item ID for the user’s current subscription.
+- `purchaseToken`: This is the purchase token for the user’s current subscription. Like it was noted earlier, it’s a good idea to keep track of the purchase tokens in your backend. And for this scenario and others, you should associate a user to their current purchases and purchase tokens as well.
+- `prorationMode`: This is how the new subscription will be charged when it replaces the user’s current subscription.
+
+| Proration Mode                                | Description                                                                                                                                                                                                                            |
+| --------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `"immediateAndChargeProratedPrice"`           | The subscription is upgraded immediately, and the billing cycle remains the same. The price difference for the remaining period is then charged to the user.                                                                           |
```

Comment:
Right now, the built table is very hard to read because the `code` block wraps stuff pretty aggressively. I recommend removing the code ticks around these items (maybe the quotes too?) to fix the table balancing

Reply:
done!